client/editor/tool_tile.cpp

```diff
@@ -86,17 +137,224 @@ void editor_tool_tile::close_tool()
+}
+
+/**
+ * \brief Return a string of the final road on a tile
+ */
+const QString editor_tool_tile::get_tile_road_name(struct tile *ptile)
+{
+  QString sroad = nullptr;
+  extra_type_by_cause_iterate(EC_ROAD, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sroad = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (sroad != nullptr) {
+    return sroad;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a strong of the added infrastructure on a tile
+ */
+const QString editor_tool_tile::get_tile_infra_name(struct tile *ptile)
+{
+  QString sinfra = nullptr;
+  extra_type_by_cause_iterate(EC_IRRIGATION, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sinfra = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  extra_type_by_cause_iterate(EC_MINE, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sinfra + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (sinfra != nullptr) {
+    return sinfra;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a strong of the any nuisances on a tile
+ */
+const QString editor_tool_tile::get_tile_nuisance_name(struct tile *ptile)
+{
+  QString snuisance = nullptr;
+  extra_type_by_cause_iterate(EC_POLLUTION, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      snuisance = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  extra_type_by_cause_iterate(EC_FALLOUT, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      snuisance + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (snuisance != nullptr) {
+    return snuisance;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a string of any huts on a tile
+ */
+const QString editor_tool_tile::get_tile_hut_name(struct tile *ptile)
+{
+  QString shut = nullptr;
+  extra_type_by_cause_iterate(EC_HUT, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      shut + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (shut != nullptr) {
+    return shut;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a string of any bases on a tile
+ */
+const QString editor_tool_tile::get_tile_base_name(struct tile *ptile)
+{
+  QString sbase = nullptr;
+  extra_type_by_cause_iterate(EC_BASE, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sbase + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (sbase != nullptr) {
+    return sbase;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a pixmap of the selected tile.
+ * TODO: Add more layers over time, right now we show the base tile and any
+ * resource We don't show any other infrastructure extras, such as
+ * roads/rivers, mines, irrigation, bases, or pollution.
+ */
+QPixmap editor_tool_tile::get_tile_sprites(struct tile *ptile,
```

Comment:
```suggestion
QPixmap editor_tool_tile::get_tile_sprite(const tile *ptile) const
```
I guess `terrain` is always `tile_terrain(ptile)` so we don't need it. It also returns a single pixmap.